pkg/querier/blocks_store_queryable.go

```diff
@@ -504,30 +502,23 @@ func (q *blocksStoreQuerier) selectSorted(sp *storage.SelectHints, matchers ...*
 		resWarnings)
 }
 
-func (q *blocksStoreQuerier) queryWithConsistencyCheck(ctx context.Context, logger log.Logger, minT, maxT int64, shard *sharding.ShardSelector,
-	queryFunc func(clients map[BlocksStoreClient][]ulid.ULID, minT, maxT int64) ([]ulid.ULID, error)) error {
-	// If queryStoreAfter is enabled, we do manipulate the query maxt to query samples up until
-	// now - queryStoreAfter, because the most recent time range is covered by ingesters. This
-	// optimization is particularly important for the blocks storage because can be used to skip
-	// querying most recent not-compacted-yet blocks from the storage.
-	if q.queryStoreAfter > 0 {
-		now := time.Now()
-		origMaxT := maxT
-		maxT = math.Min(maxT, util.TimeToMillis(now.Add(-q.queryStoreAfter)))
-
-		if origMaxT != maxT {
-			level.Debug(logger).Log("msg", "the max time of the query to blocks storage has been manipulated", "original", origMaxT, "updated", maxT)
-		}
+type _queryFunc func(clients map[BlocksStoreClient][]ulid.ULID, minT, maxT int64) ([]ulid.ULID, error)
 
-		if maxT < minT {
-			q.metrics.storesHit.Observe(0)
-			level.Debug(logger).Log("msg", "empty query time range after max time manipulation")
-			return nil
-		}
+func (q *blocksStoreQuerier) queryWithConsistencyCheck(
+	ctx context.Context, logger log.Logger, minT, maxT int64, shard *sharding.ShardSelector, queryFunc _queryFunc,
+) error {
+
+	if !QueryBlockStore(q.queryStoreAfter, time.Now(), minT) {
+		q.metrics.storesHit.Observe(0)
+		level.Debug(logger).Log("msg", "empty query time range after max time manipulation")
```

Comment:
[nit] I know this was the message we logged before for this situation, but perhaps something like "not querying blocks store because the query time range does not start before the 'query store after' threshold" would be clearer?

Reply:
updated to `not querying block store; query time range begins after the query-store-after limit`